editor/scene/2d/canvas_item_editor_gizmos.cpp

```diff
@@ -0,0 +1,810 @@
+	ClassDB::bind_method(D_METHOD("add_polygon", "points", "color"), &EditorCanvasItemGizmo::add_polygon, DEFVAL(Color(1, 1, 1)));
+	ClassDB::bind_method(D_METHOD("add_polyline", "points", "color"), &EditorCanvasItemGizmo::add_polyline, DEFVAL(Color(1, 1, 1)));
+	ClassDB::bind_method(D_METHOD("add_rect", "rect", "color"), &EditorCanvasItemGizmo::add_rect, DEFVAL(Color(1, 1, 1)));
+	ClassDB::bind_method(D_METHOD("add_collision_segments", "segments"), &EditorCanvasItemGizmo::add_collision_segments);
+	ClassDB::bind_method(D_METHOD("add_collision_rect", "rect"), &EditorCanvasItemGizmo::add_collision_rect);
+	ClassDB::bind_method(D_METHOD("add_collision_polygon", "polygon"), &EditorCanvasItemGizmo::add_collision_polygon);
+	ClassDB::bind_method(D_METHOD("add_handles", "handles", "color", "ids", "secondary"), &EditorCanvasItemGizmo::add_handles, DEFVAL(false));
+	ClassDB::bind_method(D_METHOD("set_canvas_item", "canvas_item"), &EditorCanvasItemGizmo::_set_canvas_item);
+	ClassDB::bind_method(D_METHOD("get_canvas_item"), &EditorCanvasItemGizmo::get_canvas_item);
+	ClassDB::bind_method(D_METHOD("get_plugin"), &EditorCanvasItemGizmo::get_plugin);
+	ClassDB::bind_method(D_METHOD("clear"), &EditorCanvasItemGizmo::clear);
+	ClassDB::bind_method(D_METHOD("set_hidden", "hidden"), &EditorCanvasItemGizmo::set_hidden);
+	ClassDB::bind_method(D_METHOD("is_subgizmo_selected", "id"), &EditorCanvasItemGizmo::is_subgizmo_selected);
+	ClassDB::bind_method(D_METHOD("get_subgizmo_selection"), &EditorCanvasItemGizmo::get_subgizmo_selection);
+
+	GDVIRTUAL_BIND(_redraw);
+	GDVIRTUAL_BIND(_get_handle_name, "id", "secondary");
+	GDVIRTUAL_BIND(_is_handle_highlighted, "id", "secondary");
+
+	GDVIRTUAL_BIND(_get_handle_value, "id", "secondary");
+	GDVIRTUAL_BIND(_begin_handle_action, "id", "secondary");
+	GDVIRTUAL_BIND(_set_handle, "id", "secondary", "point");
+	GDVIRTUAL_BIND(_commit_handle, "id", "secondary", "restore", "cancel");
+
+	GDVIRTUAL_BIND(_subgizmos_intersect_point, "point");
+	GDVIRTUAL_BIND(_subgizmos_intersect_rect, "rect");
+	GDVIRTUAL_BIND(_get_subgizmo_transform, "id");
+	GDVIRTUAL_BIND(_set_subgizmo_transform, "id", "transform");
+	GDVIRTUAL_BIND(_commit_subgizmos, "ids", "restores", "cancel");
+}
+
+EditorCanvasItemGizmo::EditorCanvasItemGizmo() {
+	valid = false;
+	hidden = false;
+	selected = false;
+	canvas_item = nullptr;
+	gizmo_plugin = nullptr;
+}
+
+ EditorCanvasItemGizmo::~EditorCanvasItemGizmo() {
+	if (gizmo_plugin != nullptr) {
+		gizmo_plugin->unregister_gizmo(this);
+	}
+	clear();
+}
+
+/////
+
+String EditorCanvasItemGizmoPlugin::get_gizmo_name() const {
+	String ret;
+	if (GDVIRTUAL_CALL(_get_gizmo_name, ret)) {
+		return ret;
+	}
+	WARN_PRINT_ONCE("A CanvasItem editor gizmo has no name defined (it will appear as \"Unnamed Gizmo\"  in the \"View > Gizmos\" menu). To resolve this, override the `_get_gizmo_name()` function to return a String in the script that extends EditorCanvasItemGizmoPlugin.");
```

Comment:
```suggestion
	WARN_PRINT_ONCE("A CanvasItem editor gizmo has no name defined (it will appear as \"Unnamed Gizmo\" in the \"View > Gizmos\" menu). To resolve this, override the `_get_gizmo_name()` function to return a String in the script that extends EditorCanvasItemGizmoPlugin.");
```